sdk/core/core-rest-pipeline/src/interfaces.ts

```diff
@@ -47,6 +47,101 @@ export interface HttpHeaders extends Iterable<[string, string]> {
   toJSON(options?: { preserveCase?: boolean }): RawHttpHeaders;
 }
 
+/**
+ * A blob that is streamable. `Blob` and `File` satisfy this interface.
+ */
+export interface StreamableBlob {
+  /**
+   * A stream (or a function that returns a stream) representing the blob's contents.
+   */
+  stream: ReadableStream | NodeJS.ReadableStream | (() => ReadableStream | NodeJS.ReadableStream);
+
+  /**
+   * The MIME type of the blob.
+   */
+  type?: string;
+
+  /**
+   * The size of the blob, if known.
+   */
+  size?: number;
+}
+
+/**
+ * Blob with content as a Uint8Array
+ */
+export interface InMemoryBlob {
+  /**
+   * The content of the blob as a Uint8Array
+   */
+  content: Uint8Array;
+
+  /**
+   * The MIME type of the blob.
+   */
+  type?: string;
+}
+
+/**
+ * A type representing a blob.
+ *
+ * `Blob` and `File` objects satisfy this, but a ReadableStream (web or Node) can be
+ * used instead by creating an object with the `stream` property set to the stream.
+ */
+export type BlobLike = StreamableBlob | InMemoryBlob;
+
+/**
+ * A type representing a file to be uploaded.
+ * The `File` class satisfies this type, allowing for instances to be passed directly. If the file to be
+ * uploaded is not represented by a `File` or `Blob`, an object containing a stream or content buffer and optional file type and file name
+ * can be used instead.
+ */
+export type FileLike = BlobLike & {
```

Comment:
Could we update this to be an interface instead?

Reply:
No, `BlobLike` is a union so this must be a type alias.